### PR TITLE
fixes urls

### PR DESCRIPTION
**Pull request by Nowosad**

### CODE REVIEW

File: 01-introduction.Rmd [hunk 1]
```diff
@@ -15,7 +15,7 @@ By the end of the book, we hope you will be ready to apply your skills tackle re
 
 Over the last few decades free and open source software for geospatial (FOSS4G\index{FOSS4G}) has progressed at an astonishing rate.
 Thanks to organizations such as OSGeo, advanced geographic techniques are no longer the preserve of those with expensive hardware and software: anyone can now download and run high-performance software for geocomputation.
-Open source Geographic Information Systems (GIS\index{GIS}), such as [QGIS](http://qgis.org/en/site/)\index{QGIS}, have made geographic analysis accessible worldwide.
+Open source Geographic Information Systems (GIS\index{GIS}), such as [QGIS](https://qgis.org/en/site/)\index{QGIS}, have made geographic analysis accessible worldwide.
```
Comment (Robinlovelace) [inline 01-introduction.Rmd:18]: :+1:

File: 01-introduction.Rmd [hunk 2]
```diff
@@ -186,14 +186,14 @@ The use of R code, therefore, enables teaching geocomputation with reference to
...
 
 An important feature of R (and Python) is that it is an interpreted language.
 This is advantageous because it enables interactive programming in a Read–Eval–Print Loop (REPL):\index{REPL}
 code entered into the console is immediately executed and the result is printed, rather than waiting for the intermediate stage of compilation.
 On the other hand, compiled languages such as C++\index{C++} and Java\index{Java} tend to run faster (once they have been compiled).
 
-C++\index{C++} provides the basis for many GIS packages such as [QGIS](https://www.qgis.org/)\index{QGIS}, [GRASS](https://grass.osgeo.org/)\index{GRASS} and [SAGA](http://www.saga-gis.org/)\index{SAGA} so it is a sensible starting point.
+C++\index{C++} provides the basis for many GIS packages such as [QGIS](https://www.qgis.org/en/site/)\index{QGIS}, [GRASS](https://grass.osgeo.org/)\index{GRASS} and [SAGA](https://saga-gis.sourceforge.io/)\index{SAGA} so it is a sensible starting point.
```
Comment (Robinlovelace) [inline 01-introduction.Rmd:196]: Unusual website for landing page but seems legit!

File: 02-spatial-data.Rmd [hunk 3]
```diff
@@ -8,7 +8,7 @@
 This is the first practical chapter of the book, and therefore it comes with some software requirements.
 <!-- toDo: rl-->
 <!-- should we be that specific regarding the r version?-->
-You need access to a computer with a recent version of R installed (R [4.2.0](https://stat.ethz.ch/pipermail/r-announce/2022/000683.html) or a later version).
+You need access to a computer with a recent version of R installed (R [4.3.0](https://stat.ethz.ch/pipermail/r-announce/2022/000683.html) or a later version).
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd:11]: Big +1

File: 08-read-write-plot.Rmd [hunk 4]
```diff
@@ -467,13 +467,13 @@ tmap_arrange(tm1, tm2)
...
 Naturally, some options are specific to certain drivers.^[
-A list of supported vector formats and options can be found at http://gdal.org/ogr_formats.html.
+A list of supported vector formats and options can be found at https://gdal.org/ogr_formats.html.
 ]
 For example, think of coordinates stored in a spreadsheet format (`.csv`).
 To read in such files as spatial objects, we naturally have to specify the names of the columns (`X` and `Y` in our example below) representing the coordinates.
 We can do this with the help of the `options` parameter.
 To find out about possible options, please refer to the 'Open Options' section of the corresponding GDAL\index{GDAL} driver description.
-For the comma-separated value (csv) format, visit http://www.gdal.org/drv_csv.html.
+For the comma-separated value (csv) format, visit https://gdal.org/drv_csv.html.
```
Comment (Robinlovelace) [inline 08-read-write-plot.Rmd:476]: :+1: our old URLs were very out of date!

**Review by Robinlovelace — APPROVED** [review]
Great job @Nowosad

File: 13-transport.Rmd [hunk 5]
```diff
@@ -513,7 +513,7 @@ Routing can be time and memory-consuming, resulting in large objects, due to the
 We will therefore filter the desire lines before calculating routes in this section.
 
 Cycling is most beneficial when it replaces car trips.
-Short (around 5 km, which can be cycled in 15 minutes at a speed of 20 km/hr) have a relatively high probability of being cycled, and the maximum distance increases when trips are made by [electric bike](https://www.sciencedirect.com/science/article/pii/S0967070X21003401) [@lovelace_propensity_2017].
+Short (around 5 km, which can be cycled in 15 minutes at a speed of 20 km/hr) have a relatively high probability of being cycled, and the maximum distance increases when trips are made by [electric bike](https://doi.org/10.1016/j.tranpol.2021.11.019) [@lovelace_propensity_2017].
```
Comment (Robinlovelace) [inline 13-transport.Rmd:516]: Marginal whether we need  that  URL and arguably should be reference instead if we do keep it for consistency. My bad!

**Review by Robinlovelace — APPROVED** [review]

File: 13-transport.Rmd (hunk 5 above)
Comment (Nowosad) [reply 13-transport.Rmd:516]: This url was found broken by the automatic system. But as you said, maybe it is not even needed and the reference could be better here.

Comment (Robinlovelace) [reply 13-transport.Rmd:516]: I suggest remove the link. There are loads of studies on E-bikes.

File: 01-introduction.Rmd [hunk 6]
```diff
@@ -401,9 +401,9 @@ This includes the **mapsf** package (successor of **cartography**) [@giraud_maps
 
 <!-- spatstat?? -->
 
-In late 2021, the planned retirement of **rgdal**, **rgeos** and **maptools** at the end of 2023 was announced on [the R-sig-Geo mailing list](https://stat.ethz.ch/pipermail/r-sig-geo/2021-September/028760.html) by Roger Bivand.
-This would have a large impact on existing workflows applying these packages, but also will influence the packages that depend on **rgdal**, **rgeos** or **maptools**. 
-Therefore, Bivand's suggestion is to transition to more modern tools, including **sf** and **terra**, as explained in this book's next chapters.
+In late 2021, the planned retirement of **rgdal**, **rgeos** and **maptools** was announced on [the R-sig-Geo mailing list](https://stat.ethz.ch/pipermail/r-sig-geo/2021-September/028760.html) by Roger Bivand.
+This retirement at the end of 2023 have had a large impact on existing workflows applying these packages, but also will influence the packages that depend on **rgdal**, **rgeos** or **maptools**.
```
Comment (Robinlovelace) [inline 01-introduction.Rmd:405]:
```suggestion
These packages, and packages that depended on them, were removed from CRAN at the end of 2023.
```

Comment (Robinlovelace) [reply 01-introduction.Rmd:405]: There's repetition of `**rgdal**, **rgeos** and **maptools**`.

The typo fix option: 

This retirement at the end of 2023 **has** had ... 

But I think less is more now that they have been removed.

**Comment by Nowosad** [comment]
I am merging it now (not to mess with the repo). I think the remaining issues at https://github.com/geocompx/geocompr/issues/991 can be fixes directly in the main.

**Review by Robinlovelace — COMMENTED** [review]
🚀

**Pull request merged**